### ISSUE TITLE
Separate Shaded Jar for Exporter Auto

### ISSUE THREAD

**Issue by dansiviter** [GoogleCloudPlatform/opentelemetry-operations-java#161]
There is only a shaded JAR for `exporter-auto` which has not moved the package for the bulk of `io.grpc` classes. Now I get issues if I use a API incompatible gRPC version:

```
java.lang.AbstractMethodError: Receiver class io.grpc.netty.shaded.io.grpc.netty.NettyClientTransport does not define or inherit an implementation of the resolved method 'abstract io.grpc.internal.ClientStream newStream...
```

Please split this into `exporter-auto` and `exporter-auto-shaded`, similar to `grpc-netty` and `grpc-netty-shaded` to avoid these situations.

**Comment by aabmass** [comment]
@psx95 is taking this on

**Comment by punya** [comment]
Notes from team discussion:

1. We should make the default artifact *not* shadowed.
2. We should update our documentation to point people at the shadowed jar for agent + in-process use cases (with explicit instructions for each). There should be an "advanced" section detailing when someone might want to use the un-shadowed version, and how.
3. We should shadow *all* implementation dependencies. (Right now it's just an uberjar, but we want the classes to be placed under a common prefix too -- just like the resource mapping project does right now.)

**Comment by psx95** [comment]
#### Diagnosis

When using the shaded-jar *(with relocated dependencies)*, there have been some issues running the examples provided within this repository in a containerized environment *(like GKE; issue was reproducible locally via docker containers)*.

The crux of the issue was that `google-cloud-monitoring` which is a transitive dependency of `exporter-auto` has a depedency on [conscrypt](https://github.com/google/conscrypt). In a containerized environment, classes from this package are required, which are now relocated in the shadowJar and so we get a ClassNotFoundError. This is a known issue with the repository - see [here](https://github.com/google/conscrypt/issues/811)

This issue has a workaround listed but it is not feasible for us in this case. 

#### Next Steps
Since the workaround listed is not feasible, we can skip the shading for *conscrypt* library - Since the team decided on shading all transitive dependencies - *not just gRPC*, shadow's automatic relocation could be used on this. 

As part of this, it would be better to first upgrade the gradle plugin and consecutively, the shadow plugin to **v8.0.0+**.

**Comment by psx95** [comment]
Blocked on #210

**Comment by ViliusS** [comment]
For me it's still not clear why Google OT exporter packages are uber jars and why shading is necessary at all. Is this really a good practice?

**Comment by ViliusS** [comment]
> Since shading/creating uberJar dramatically increases the size of the resultant JAR, we plan to provide both variants (_uberJar as well as normal JAR_) and let the users choose which one to use.

Thank you for the detailed explanation, I was not aware of the javaagent requirement.

Providing separate uber and normal JARs would be best indeed. Will be waiting for normal exporter-auto jar for our use cases.

**Comment by psx95** [comment]
Until the #215 is completed, a shaded JAR for exporter auto can be obtained by running - 
```
./gradlew shadowJar
```
against the HEAD of the repo. 

The shaded JAR can be found in `/exporters/auto/build/libs/exporter-auto-0.1.0-SNAPSHOT-shaded.jar`
